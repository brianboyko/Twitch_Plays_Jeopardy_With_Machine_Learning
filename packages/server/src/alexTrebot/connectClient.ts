import yaml from "yaml";
import fs from "fs";
import { ChatUserstate, Client } from "tmi.js";
import config from "../config";
import makeHandler from "./handler";
import { Server } from "socket.io";
import { wsServer } from "../sockets/commands";

export const getText = (): Promise<Record<string, Array<string>>> =>
  new Promise((resolve, reject) => {
    fs.readFile("./text.yaml", "utf8", (err, data) => {
      if (err) {
        console.error(err);
        reject(err);
      }
      resolve(yaml.parse(data));
    });
  });

export const connectClient = async (io: Server): Promise<Client> => {
  const client = Client({
    options: { debug: true },
    connection: {
      secure: true,
      reconnect: true,
    },
    channels: [config.CHANNEL_NAME],
    identity: {
      username: config.BOT_USERNAME,
      password: config.OAUTH_TOKEN,
    },
  });
  const text = await getText();
  const handler = await makeHandler(io, client, text);
  client.on("connected", (addr, port) => {
    console.info(`* Connected to ${addr}:${port}`);
  });
  client.on(
    "message",
    (
      target: string,
      context: ChatUserstate,
      message: string,
      isSelf: boolean
    ) => {
      io.emit(
        wsServer.CHAT_LOG,
        JSON.stringify({ message, user: context["display-name"] })
      );
      handler(target, context, message, isSelf);
    }
  );
  client.connect();
  console.info("alextrebot is connected");
  return client;
};

export default connectClient;
