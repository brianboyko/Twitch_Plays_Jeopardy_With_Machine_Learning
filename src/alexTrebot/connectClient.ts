import { ChatUserstate, Client } from 'tmi.js';
import config from '../config';

export const connectClient = () => {
  const client = Client({
    options: { debug: true },
    connection: {
      secure: true,
      reconnect: true,
    },
    channels: [config.CHANNEL_NAME],
    identity: {
      username: config.BOT_USERNAME,
      password: config.OAUTH_TOKEN,
    },
  });
  client.on('connected', (addr, port) => {
    console.log(`* Connected to ${addr}:${port}`);
  });
  client.on('message', (target, context: ChatUserstate, message: string, isSelf: boolean) => {
    if (isSelf) {
      return;
    }
    console.log(`${context['display-name']}: ${message}`);

    if (['!whois', '!whatis', '!whenis', '!whereis'].some((key: string) => message.startsWith(key))) {
      client.say(target, `What is: ${message}`);
    }
  });
  client.connect();
  return client;
};

export default connectClient;
