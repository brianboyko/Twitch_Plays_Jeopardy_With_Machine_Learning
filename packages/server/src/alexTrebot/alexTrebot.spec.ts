import { getText } from "./connectClient";

test("getText gets the text", async () => {
  const text = await getText();
  expect(text).toMatchSnapshot();
});
