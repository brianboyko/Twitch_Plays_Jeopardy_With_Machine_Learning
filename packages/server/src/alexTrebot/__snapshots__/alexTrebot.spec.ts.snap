// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`getText gets the text 1`] = `
Object {
  "ai": Array [
    "The big issue with a game like Jeopardy is that a computer makes a poor Jeopardy judge. We're trying to do our best by using machine learning (Tensorflow) in order to get our AI, \\"Art Ificial Fleming\\" to get better at recognizing correct and incorrect responses.",
    "For example, computers might have a real problem recognizing that \\"!whois Henry Tudor\\" and \\"!whois Henry the 8th\\" are both acceptable answers for \\"!whois Henry VIII\\"",
    "Soon, we will have a website where human judges will be able to make real-time or near-real time corrections to what our AI does, (awarding points in those corrections) and hopefully (crosses fingers), Art will learn from them and get better over time, requiring less human intervention. In the meantime, please forgive Art's mistakes.  This is an in-dev, solo project.",
  ],
  "help": Array [
    "Welcome to JeopardAI! v.0.0.0.0.001 inDev.  To start, make sure to open up a whisper window to alextrebot.",
    "-- Starting a game - \\"!startgame [seed]\\" - Starts a game. Leave 'seed' blank if you want a random game, or enter any string for a seeded game.",
    "-- Registering to play - \\"!register\\". Late registration is allowed.",
    "-- Selecting a clue - \\"!select [category keyword] [value]\\" (if you have control of the board) The keyword is the CAPITALIZED word. E.g. in \\"BIBLICAL who's who\\", type \\"!select biblical 800\\" to select the $800 clue.",
    "-- Responding to a clue - \\"/w alextrebot !whatis [your response]\\" (or \\"!whatis [your response]\\" if in alextrebot's whisper window. \\"!whois\\", \\"!whereis\\", & \\"!whenis\\" also work.",
    "-- Wagering - \\"/w alextrebot !wager [amount]\\" These are secret in Final Jeopardy but public in Daily Doubles.",
    "-- Learn More - \\"!ai\\" to learn more about our machine learning jeopardy judge, \\"Art Ificial Fleming\\" and how you can help A.I. Fleming learn by becoming a judge.",
    "-- Legal & Privacy - \\"!legal\\" - disclaimer & privacy policy.",
    "For more info, go to www.websiteToCome.com/underconstruction",
  ],
  "legal": Array [
    "This project, site, twitch stream, etc. are not affiliated with Jeopardy!, King World television, Sony Television, the estates of Alex Trebek or Art Fleming, or IBM.",
    "This project is non-commercial and being used for learning purposes.",
    "Questions have been retrieived from J-Archive.com",
    "As of now, no user data, identifiable or otherwise, is saved on our servers. This will be updated if it changes.",
  ],
}
`;
